### DISCUSSION TITLE
How to make Outline listen to smth other than localhost&

### DISCUSSION THREAD

**Discussion by Comandante1977** [outline/outline#3666] [Self Hosting] [answered]
Hi, everybody!

Just installed outline on server without docker. Everything seems working, but outline listen on localhost.
I have reverse proxy on other server, so I have to point to ip address of outline.
Is there any solution or I have to install reverse proxy on outline server in any way?

Thanks.

**Comment by almereyda** [comment, marked as answer]
It appears this setting is hard coded.

You could omit running the reverse proxy on the Outline machine, with using a tiny `socat`/`netcat` pipe.

Alternatively, the Docker distribution allows you to customise exposed ports in more detail.

**Reply by Comandante1977** [reply]
I see. Good idea about pipe! Thanks. Will try tomorrow.

**Reply by almereyda** [reply]
Glad to hear that. Please also have a look at https://unix.stackexchange.com/questions/311492/redirect-application-listening-on-localhost-to-listening-on-external-interface on how to make this work, either via IPTables, with `socat` for IPv4 or with `socat` for IPv6, too.
